osu.Desktop/OsuGameDesktop.cs

```diff
@@ -54,19 +56,29 @@ public override void SetHost(GameHost host)
         private void dragDrop(DragEventArgs e)
         {
             // this method will only be executed if e.Effect in dragEnter gets set to something other that None.
-            var dropData = e.Data.GetData(DataFormats.FileDrop) as object[];
+            var dropData = (object[])e.Data.GetData(DataFormats.FileDrop);
             var filePaths = dropData.Select(f => f.ToString()).ToArray();
-            ImportBeatmapsAsync(filePaths);
+
+            if (filePaths.All(f => Path.GetExtension(f) == @".osz"))
+                Task.Run(() => BeatmapDatabase.Import(filePaths));

```

Comment:
Not sure if we want to support this. I guess it might make sense if you drag and osz and osr at the same time (where they complement each other) but maybe not so much when two different actions could result from the same drag action.

Wanna keep the workflow here simple for the time being.